src/Tooltip.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { CSSProperties, forwardRef, memo } from "react";
+import type { Equals } from "tsafe";
+import { assert } from "tsafe/assert";
+import { useAnalyticsId } from "./tools/useAnalyticsId";
+import { fr } from "./fr";
+import { cx } from "./tools/cx";
+
+export type UseTooltipParams = UseTooltipParams.Common &
+    (UseTooltipParams.WithTriggerOnClick | UseTooltipParams.WithTriggerOnHover);
+
+export namespace UseTooltipParams {
+    export type Common = {
+        id?: string;
+        className?: string;
+        trigger?: "click" | "hover";
+        buttonLabel?: string;
+        style?: CSSProperties;
+    };
+
+    export type WithTriggerOnClick = {
+        trigger: "click";
+        buttonLabel?: string;
+    };
+
+    export type WithTriggerOnHover = {
+        trigger: "hover";
+        buttonLabel: never;
+    };
+}
+
+type UseTooltipReturn = {
+    tooltipOwnerProps: {
+        "aria-describedby": string;
+        className?: string;
+    };
+    TooltipOwnerButtonComponent?: JSX.Element;
+    TooltipComponent: ({ children }: { children: string }) => JSX.Element;
+};
+
+export const useTooltip = (params?: UseTooltipParams): UseTooltipReturn => {
+    const tooltipId = useAnalyticsId({
+        "defaultIdPrefix": "fr-tooltip",
+        "explicitlyProvidedId": params?.id
+    });
+
+    return {
+        tooltipOwnerProps: {
+            "aria-describedby": tooltipId,
+            ...(params?.trigger === "click" && {
+                "className": cx(fr.cx("fr-btn--tooltip"), params?.className)
+            })
+        },
+        TooltipOwnerButtonComponent:
+            params?.trigger === "click" ? (
+                <TooltipButton tooltipId={tooltipId} label={params.buttonLabel} />
+            ) : undefined,
+        TooltipComponent: ({ children }) => (
+            <Tooltip {...params} id={tooltipId}>
+                {children}
+            </Tooltip>
+        )
+    };
+};
+
+const TooltipButton = memo(
+    forwardRef<HTMLButtonElement, { tooltipId: string; label?: string }>((props, ref) => {
+        const { tooltipId, label, ...rest } = props;
+        assert<Equals<keyof typeof rest, never>>();
+
+        const tooltipButtonId = `${tooltipId}-button`;
+
+        return (
+            <button
+                ref={ref}
+                className={fr.cx("fr-btn--tooltip", "fr-btn")}
+                aria-describedby={tooltipId}
+                id={tooltipButtonId}
+            >
+                {label ?? "En savoir plus"}
```

Comment:
Default label should be "Information contextuelle", as per https://www.systeme-de-design.gouv.fr/elements-d-interface/composants/infobulle